src/llama-model.cpp

```diff
@@ -13198,6 +13203,8 @@ llama_memory_i * llama_model::create_memory(const llama_memory_params & params,
     llama_memory_i * res;
 
     switch (arch) {
+        // Models that need specific instantiation should be handled in the
```

Comment:
This is a slight architectural refactor to avoid the need to duplicate the arch switch statement between this and `llm_arch_is_[hybrid]_recurrent`. I'm not strictly tied to this, but it feels a little cleaner to hold a single source-of-truth for determining which models are hybrid/hybrid-recurrent.